pkg/console/console.go

```diff
@@ -126,46 +127,55 @@ func generateConsoleCSPString(config *Config, nonce string) string {
 	return webui.GenerateCSPString(cspMap)
 }
 
+func cspHeader(console *Console) webmiddleware.MiddlewareFunc {
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			r, nonce := webui.WithNonce(r)
+			cspString := generateConsoleCSPString(console.configFromContext(r.Context()), nonce)
+			w.Header().Set("Content-Security-Policy", cspString)
+			next.ServeHTTP(w, r)
+		})
+	}
+}
+
+func templateMiddleware(console *Console) webmiddleware.MiddlewareFunc {
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			config := console.configFromContext(r.Context())
+			r = webui.WithTemplateData(r, config.UI.TemplateData)
+			frontendConfig := config.UI.FrontendConfig
+			frontendConfig.Language = config.UI.TemplateData.Language
+			r = webui.WithAppConfig(r, struct {
+				FrontendConfig
+			}{
+				FrontendConfig: frontendConfig,
+			})
+			next.ServeHTTP(w, r)
+		})
+	}
+}
+
 // RegisterRoutes implements web.Registerer. It registers the Console to the web server.
 func (console *Console) RegisterRoutes(server *web.Server) {
-	group := server.Group(
-		console.config.Mount,
-		func(next echo.HandlerFunc) echo.HandlerFunc {
-			return func(c echo.Context) error {
-				nonce := webui.GenerateNonce()
-				c.Set("csp_nonce", nonce)
-				cspString := generateConsoleCSPString(console.configFromContext(c.Request().Context()), nonce)
-				c.Response().Header().Set("Content-Security-Policy", cspString)
-				return next(c)
-			}
-		},
-		func(next echo.HandlerFunc) echo.HandlerFunc {
-			return func(c echo.Context) error {
-				config := console.configFromContext(c.Request().Context())
-				c.Set("template_data", config.UI.TemplateData)
-				frontendConfig := config.UI.FrontendConfig
-				frontendConfig.Language = config.UI.TemplateData.Language
-				c.Set("app_config", struct {
-					FrontendConfig
-				}{
-					FrontendConfig: frontendConfig,
-				})
-				return next(c)
-			}
-		},
-		web_errors.ErrorMiddleware(map[string]web_errors.ErrorRenderer{
-			"text/html": webui.Template,
-		}),
-		middleware.CSRF("_console_csrf", console.config.Mount, console.GetBaseConfig(console.Context()).HTTP.Cookie.HashKey),
+	consoleRouter := server.Router().NewRoute().Subrouter()
```

Comment:
You can use `server.Router().NewRoute().Path(console.config.Mount).Subrouter()`, instead of register the router as its own handler.